system-int-test/test/cmr/system_int_test/ingest/subscription_processing_test.clj

```diff
@@ -49,71 +40,305 @@
+           _ (index/wait-until-indexed)
+         ;; Setup subscriptions
+           _sub1 (subscription-util/create-subscription-and-index coll1 "test_sub_prov1_coll1" "user2" "day_night_flag=day")
+           _sub2 (subscription-util/create-subscription-and-index coll2 "test_sub_prov1_coll2" "user2" "day_night_flag=day")]
+
+       (testing "given a time constraint with no granules..."
+         (let [time-constraint "2016-01-07T00:00:00Z,2016-01-09T00:00:00Z"
+               system-context (merge (system/context) {:time-constraint time-constraint})
+               result (->> system-context
+                           (jobs/email-subscription-processing)
+                           (first)
+                           (second))]
+           (is (= (count result) 0))))
+
+       (testing "given a time constraint with only an end time, get all granules until the end time"
+         (let [time-constraint ",2016-01-04T00:00:00Z"
+               system-context (merge (system/context) {:time-constraint time-constraint})
+               result (->> system-context
+                           (jobs/email-subscription-processing)
+                           (first)
+                           (second))]
+           (is (= (count result) 3))
+           (is (= (:concept-id coll1_granule1) (:concept-id (nth result 0))))
+           (is (= (:concept-id coll1_granule2) (:concept-id (nth result 1))))
+           (is (= (:concept-id coll1_granule3) (:concept-id (nth result 2))))))
+
+       (testing "given a time constraint with only a start time, get all granules from start time until now"
+         (let [time-constraint "2016-01-02T00:00:00Z,"
+               system-context (merge (system/context) {:time-constraint time-constraint})
+               result (->> system-context
+                           (jobs/email-subscription-processing)
+                           (first)
+                           (second))]
+           (is (= (count result) 3))
+           (is (= (:concept-id coll1_granule2) (:concept-id (nth result 0))))
+           (is (= (:concept-id coll1_granule3) (:concept-id (nth result 1))))
+           (is (= (:concept-id coll1_granule4) (:concept-id (nth result 2))))))
+
+       (testing "given a time constraing with a start and end time, get the granules that fall within the time constraint"
+         (let [time-constraint "2016-01-02T00:00:00Z,2016-01-04T00:00:00Z"
+               system-context (merge (system/context) {:time-constraint time-constraint})
+               result (->> system-context
+                           (jobs/email-subscription-processing)
+                           (first)
+                           (second))]
+           (is (= (count result) 2))
+           (is (= (:concept-id coll1_granule2) (:concept-id (first result))))
+           (is (= (:concept-id coll1_granule3) (:concept-id (second result))))))))))
```

Comment:
I don't quite understand what functionality we are testing here, since you are bypassing the code which actually validates time constraint